code/game/objects/structures/sensor_tower.dm

```diff
@@ -61,3 +61,118 @@
 			SSminimaps.remove_marker(src)
 			SSminimaps.add_marker(src, z, MINIMAP_FLAG_ALL, "neutral_zone")
 
+/obj/structure/sensor_tower_patrol
+	name = "alpha sensor tower"
+	desc = "A tall tower with a sensor array at the top and a control box at the bottom. Has a lengthy activation process with 3 phases."
+	icon = 'icons/obj/structures/sensor.dmi'
+	icon_state = "sensor_loyalist"
+	obj_flags = NONE
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	resistance_flags = RESIST_ALL
+	var/current_timer
+	var/generate_time = 50 SECONDS
+	var/total_segments = 3 // total number of times the activation is required
+	var/activate_time = 5 SECONDS // time to start the activation
+	var/deactivate_time = 10 SECONDS // time to stop the activation proccess
+	var/completed_segments = 0 // what segment we are on, (once this hits total, sensor tower segment is finished)
+	var/id = 1
+	var/activated = FALSE // if all segments are finished
+	var/datum/aura_bearer/current_aura
```

Comment:
and anyway, it having auras is eh. They already are objectives, I don't think you need that